vyper/venom/README.md

```diff
@@ -160,3 +160,250 @@ A number of passes that are planned to be implemented, or are implemented for im
+  - Effectively translates to `JUMP` and therefore changes the program counter value.
+  - ```
+    invoke offset, label
+    ```
+- `alloca`
+  - Allocates memory of a given size at a given offset in memory.
+  - The output is the offset itself.
+  - Because the SSA form does not allow changing values of registers, handling mutable variables can be tricky. The `alloca` instruction is meant to simplify that.
+  - ```
+    out = alloca size, offset
+    ```
+- `palloca`
+  - Like the `alloca` instruction but only used for parameters of internal functions.
+  - ```
+    out = palloca size, offset
+    ```
+- `iload`
+  - Load value at immutable section of memory denoted by `offset` into `out` variable.
+  - The operand can be either a literal, which is a statically computed offset, or a variable.
+  - ```
+    out = iload offset
+    ```
+- `istore`
+  - The instruction represents a store into immutable section of memory.
+  - Like in `iload`, the offset operand can be a literal.
+  - ```
+    istore offset value
+    ```
+- phi
+  - label, variable, basic phi
+  - ```
+    %out = phi %61:2, label_a, %61, label %__main_entry
+    ```
+- `offset`
+  - Statically compute offset. Useful for `mstore`, `mload` and such.
+  - Basically `label` + `op`.
+  - ```
+    ret = offset label, op
+    ```
+- `param`
+  - The `param` instruction is used to represent function arguments passed by the stack.
+  - We assume the argument is on the stack and the `param` instruction is used to ensure we represent the argument by the `out` variable.
+  - ```
+    out = param
+    ```
+- `store`
+  - Store variable value or literal into `out` variable.
+  - ```
+    out = op
+    ```
+- dbname
+  - make and mark a data segment (one data segment in context - so maybe section it?) dunno
+- db
+  - db stores into the data segment some label? hmm
+- dloadbytes
+  - aparently the same `codecopy`-everything handled the same way. Maybe historical reasons?
+- `ret`
+  - Represents a return from an internal call.
+  - Jumps to a location given by `op`, hence modifies the program counter.
+  - ```
+    ret op
+    ```
+- exit
+  - similar like return, but jumps to one predetermined section.
+  - Used for constrcutor exit? chec why in fallback
+  - ```
+    exit
+    ```
+- sha3_64
+- `assert`
+  - Assert that `op` is zero. If it is not, revert.
+  - Calls that terminate this way do receive a gas refund.
+  - ```
+    assert op
+    ```
+- `assert_unreachable`
+  - Check that `op` is zero. If it is not, terminate.
+  - Calls that end this way do not receive a gas refund.
+  - ```
+    assert_unreachable op
+    ```
+- log
+  - topic in 0 to 4 meant for logging, translates to EVM Log0..log4 instructions
+  - ```
+    log offset, size, {topic}max4 , topic_count
+    ```
+- `nop`
+  - No operation, does nothing.
+  - ```
+    nop
+    ```
+
+### Jump instructions
+
+- `jmp`
+  - Unconditional jump to code denoted by given `label`.
+  - Changes the program counter.
```

Comment:
i don't think we actually need to say this. it depends on the architecture, or it could be optimized out (if the jumpdest is immediately following the jump)